### PR TITLE
Address Ruby dep issues and bump to `v9.0.4`

### PR DESCRIPTION
**Pull request by ianks**
Due to issues surrounding ruby-head API stability, we've had to use a web of git forks to make things work with `ruby-head`, which we continuously test in production. Now that https://github.com/oxidize-rb/rb-sys/issues/227 is fixed and [magnus v0.5.5](https://github.com/matsadler/magnus/releases/tag/0.5.5) is released, we can untangle that.

This PR does the following:

1. Upgrades magnus and rb-sys for the `stable-api` feature
2. Upgrade wasmtime to v9.0.4
3. Bump the gem the v9.0.4

After this goes through, I will release a new gem version (v9.0.4).

### CODE REVIEW

File: ext/Cargo.toml [hunk 1]
```diff
@@ -16,20 +16,31 @@ ruby-api = []
 
 [dependencies]
 lazy_static = "1.4.0"
-magnus = { version = "0.5.3", features = ["rb-sys-interop"] }
-rb-sys = "~0.9.81"
-wasmtime = { version = "= 9.0.1" }
-wasmtime-wasi = "= 9.0.1"
-wasi-common = "= 9.0.1"
-wasi-cap-std-sync = "= 9.0.1"
+# For ruby-head support, we have to lock to 0.5.4 until
```
Comment (saulecabrera) [inline ext/Cargo.toml:19]: Should this comment be updated to say `0.5.5`?

File: ext/Cargo.toml [hunk 2]
```diff
@@ -16,20 +16,31 @@ ruby-api = []
...
+wasmtime = { version = "= 9.0.4" }
+wasmtime-wasi = "= 9.0.4"
+wasi-common = "= 9.0.4"
+wasi-cap-std-sync = "= 9.0.4"
 cap-std = "1.0.5"
 anyhow = "*" # Use whatever Wasmtime uses
-wat = "1.0.69"
-tokio = { version = "1.28.2", features = ["rt", "rt-multi-thread", "time", "net"], optional = true }
-async-timer = { version = "1.0.0-beta.8", features = ["tokio1"], optional = true }
+wat = "1.0.63"
```
Comment (saulecabrera) [inline ext/Cargo.toml:31]: Any reason why we went back to `1.0.63`?

File: ext/Cargo.toml [hunk 3]
```diff
@@ -16,20 +16,31 @@ ruby-api = []
...
+tokio = { version = "1.28.1", features = [
```
Comment (saulecabrera) [inline ext/Cargo.toml:32]: Same here, any reason why we went back to 1.28.1?

**Review by saulecabrera — COMMENTED** [review]
Generally looks good to me. Left a couple of comments regarding the versions of a couple of dependencies.

File: lib/wasmtime/version.rb [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 module Wasmtime
-  VERSION = "9.0.1"
+  VERSION = "9.0.4.rc1"
```
Comment (jbourassa) [inline lib/wasmtime/version.rb]: What's the rationale behind a RC release?

Comment (jbourassa) [reply lib/wasmtime/version.rb]: err I see it's now 9.0.4, resolving.

**Comment by jbourassa** [comment]
Thanks Ian 🙇

File: ext/Cargo.toml [hunk 5]
```diff
@@ -16,20 +16,31 @@ ruby-api = []
...
 [dependencies]
 lazy_static = "1.4.0"
-magnus = { version = "0.5.3", features = ["rb-sys-interop"] }
-rb-sys = "~0.9.81"
-wasmtime = { version = "= 9.0.1" }
-wasmtime-wasi = "= 9.0.1"
-wasi-common = "= 9.0.1"
-wasi-cap-std-sync = "= 9.0.1"
+# For ruby-head support, we have to lock to 0.5.4 until
+# rb-sys/stable-api-compiled-fallback feature is used in Magnus.
```
[the remark this answers is not in the document]
Comment (ianks) [reply ext/Cargo.toml]: Fixed

File: ext/Cargo.toml (hunk 2 above)
Comment (ianks) [reply ext/Cargo.toml]: Git conflict resolution gone wrong, fixed now

File: ext/Cargo.toml (hunk 1 above)
Comment (ianks) [reply ext/Cargo.toml]: Fixed

**Review by jbourassa — APPROVED** [review]

**Pull request merged**